internal/httpproxy/http_proxy.go

```diff
@@ -84,6 +85,7 @@ type ProxyConn struct {
 	claims *token.GATClaims
 	timer  *time.Timer
 	mu     sync.Mutex
+	start  time.Time
```

Comment:
Not sure we need a field here. We could create a timestamp at the beginning of `authenticate()` method.